rsocket/statemachine/RSocketStateMachine.cpp

```diff
@@ -177,8 +176,6 @@ void RSocketStateMachine::resumeClient(
 
   // Disconnect a previous client if there is one.
   disconnect(std::runtime_error{"Resuming client on a different connection"});
-
-  setResumable(true);
```

Comment:
This needed to be moved up, otherwise the above `disconnect()` would always call `close()` for cold resumption.